### DISCUSSION TITLE
Support for Qwik

### DISCUSSION THREAD

**Discussion by TGTGamer** [teambit/bit#7491] [Ideas, feature requests]
Qwik is a new kind of web framework that can deliver instant loading web applications at any size or complexity. Your sites and apps can boot with about 1kb of JS (regardless of application complexity), and achieve consistent performance at scale.

Qwik is very similar to other web frameworks at a high level. Qwik is a framework that renders a tree of components resulting in an interactive application.

The unique part of Qwik is not in what it does but in how it achieves its goals. Qwik's goal is to have instant-on applications, even on mobile devices. Qwik achieves this through two main strategies:

1. Delay execution and download of JavaScript for as long as possible.
2. Serialize the execution state of the application and the framework on the server and resume it on the client.

The goal of Qwik is having only to download and execute the bare minimum of the application.

**Comment by ranm8** [comment]
Thanks for the suggestion, we are planning to build an official Quik environment in the future. Keep in mind you can build your own dev environment for Quik. Learn more at https://bit.dev/docs/getting-started/composing/dev-environments

**Reply by TGTGamer** [reply]
Gosh that too a while to respond hahaha. I've had a extensive long meeting with your `Head of Product` where we discussed the challenges you are facing with Qwik at the moment. Based on that discussion and the discussions I was told are happening at Builder.io, I've put my dreams of Qwik on hold until you guys can support it officially :)
